service/routes.py

```diff
@@ -154,10 +152,18 @@ def delete(promotion_id):
 @app.route("/promotions", methods=["GET"])
 def read_all():
     """
-    Read details of specific promotion id
+    Read details of all promotions matching search criteria
     """
-    app.logger.info("Request to Retrieve all promotions")
-    promotions = Promotion.all()
+    app.logger.info("Request to Retrieve all promotions with filters: {filters}")
+    filters = request.args.to_dict(flat=True)
+    to_list_query("promotion_scope", filters)
+    to_list_query("promotion_type", filters)
+    datetime = filters.get("datetime")
```

Comment:
Do we also need to allow users to query several values for date, name, code, and active status in one query?

Reply:
Hmm i think probably not for date, name or code?  But it's up to you! I just thought it might be useful especially for scope

Reply:
Understood. Thank you!